### PR TITLE
extract shouldRun predicate to a function in daemonset controller

### PR DESCRIPTION
**Pull request by mikedanese**
This also fixes a bug where the status is reported incorrectly since the predicates were copied in two spots and didn't match.

### CODE REVIEW

File: pkg/controller/daemon/controller.go [hunk 1]
```diff
@@ -518,6 +505,21 @@ func (dsc *DaemonSetsController) syncDaemonSet(key string) error {
 	return nil
 }
 
+func nodeShouldRunDaemonPod(node *api.Node, ds *extensions.DaemonSet) bool {
+	// Check if the node satisfies the daemon set's node selector.
+	nodeSelector := labels.Set(ds.Spec.Template.Spec.NodeSelector).AsSelector()
+	shouldRun := nodeSelector.Matches(labels.Set(node.Labels))
+	// If the daemon set specifies a node name, check that it matches with nodeName.
+	nodeName := node.Name
```
Comment (madhusudancs) [inline pkg/controller/daemon/controller.go]: Is it really required to alias node.Name? Seems to be used in only one place (in the next line).

**Comment by madhusudancs** [comment]
LGTM, other than the minor nit.

File: pkg/controller/daemon/controller.go (hunk 1 above)
Comment (mikedanese) [inline pkg/controller/daemon/controller.go]: Fixed!

**Comment by mikedanese** [comment]
Rebased.

**Comment by mikedanese** [comment]
@k8s-bot ok to test

**Comment by mikedanese** [comment]
@k8s-bot ok to test

**Comment by mikedanese** [comment]
@goolebot I signed it. @k8s-merge-robot  ok to test

**Pull request merged**